### PR TITLE
add firecache support

### PR DESCRIPTION
**Pull request by ewgenius**

### CODE REVIEW

File: img/spice-dashboard-screenshot.png
Comment (ewgenius) [inline img/spice-dashboard-screenshot.png:1]: TODO: update screenshot after finishing query editor ui

File: pkg/plugin/datasource.go [hunk 1]
```diff
@@ -43,8 +45,23 @@ func NewDatasource(settings backend.DataSourceInstanceSettings) (instancemgmt.In
 		return nil, fmt.Errorf("failed to initizlize Spice AI client: %w", err)
```
Comment (lukekim) [inline pkg/plugin/datasource.go]:
```suggestion
		return nil, fmt.Errorf("failed to initialize the Spice.xyz client: %w", err)
```

File: pkg/plugin/datasource.go [hunk 2]
```diff
@@ -43,8 +45,23 @@ func NewDatasource(settings backend.DataSourceInstanceSettings) (instancemgmt.In
...
 	}
 
+	opts, err := settings.HTTPClientOptions()
+	if err != nil {
+		return nil, fmt.Errorf("http client options: %w", err)
+	}
+	opts.Headers = map[string]string{
+		"Content-Type": "application/json",
+		"X-API-Key":    apiKey,
```
Comment (lukekim) [inline pkg/plugin/datasource.go]: Add `Accept-Encoding: gzip, deflate`

**Review by lukekim — APPROVED** [review]

File: pkg/plugin/datasource.go (hunk 2 above)
Comment (ewgenius) [reply pkg/plugin/datasource.go]: Done, also updated response parser to support new encodings

**Pull request merged**